In [7]:
# Read in the data 
import csv
with open('us-states.csv') as file:
        state_covid_data = []
        for row in csv.reader(file):
            state_covid_data.append(row)

In [21]:
# Look at the number of observations
len(state_covid_data)


10080

In [22]:
# Print the first five observations
state_covid_data[:5]

[['date', 'state', 'fips', 'cases', 'deaths'],
 ['2020-01-21', 'Washington', '53', '1', '0'],
 ['2020-01-22', 'Washington', '53', '1', '0'],
 ['2020-01-23', 'Washington', '53', '1', '0'],
 ['2020-01-24', 'Illinois', '17', '1', '0']]

### Count the number of unique dates in the data

In [24]:
# Create a set to store only the dates
dates = set()

# Loop through the list to put all the dates into the set made above
for observations in state_covid_data[1:]:
    dates.add(observations[0])

# Check the length of dates
len(dates)

225

There are 225 unique dates in the data

### Find the first date in which the District of Columbia recorded a case. 

In [39]:
# Make a list with all the state names
states = []

# Loop through the data and put all the states into the above list
for observation in state_covid_data:
    states.append(observation[1])

# use index to find the first position of District of Columbia
print(states.index("District of Columbia"))

#Use the printed index and index 0 to see the date 
print(state_covid_data[354][0])

354
2020-03-07


The first date DC appears is 2020-03-07

### Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date of its first case.

In [44]:
# Write function "firstcoviddate"
def firstcoviddate(state):
    ''' The function takes in a state name as a string, looks for its index in the already created states list, 
    and then uses that index to locate it in the original datset and to pull out the corresponding date '''
    # search the state list for the input and save the index of the first occurence
    state_index=states.index(state)
    # Use the saved index to find the observation in the dataset and save the corresponding date
    first_date=state_covid_data[state_index][0]
    return first_date

# test the functtion
firstcoviddate("District of Columbia")

'2020-03-07'

### Bonus: Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date when the number of reported cases within the state exceeded 1000.

In [43]:
def firstthousandcases():
    ''' The function runs with no arguments and prompts the user to enter a state name, then searches the dataset for that state where the number of 
    cases is greater than or equal to 1000. Once it finds that observation it prints the corresponding date then breaks'''
    overthousandcases =[]
    state = input("enter state name: ")
    for observation in state_covid_data[1:]:
        if observation[1] == state:
            if int(observation[3]) >= 1000:
                print(observation[0])
                break

#test the function
firstthousandcases()

enter state name: Michigan
2020-03-22
